### PR TITLE
Fix using Directory to move files

### PR DESCRIPTION
**Pull request by JeremyKuhne**
See #19710. NetFX behavior is that you can use Directory
to move files. This was broken trying to match other legacy
behavior in #17913. Tested on NetFX and CoreFX.

### CODE REVIEW

File: src/System.IO.FileSystem/src/System/IO/Directory.cs [hunk 1]
```diff
@@ -563,8 +563,10 @@ public static void Move(String sourceDirName, String destDirName)
             String destinationRoot = Path.GetPathRoot(destPath);
             if (!String.Equals(sourceRoot, destinationRoot, pathComparison))
                 throw new IOException(SR.IO_SourceDestMustHaveSameRoot);
-            
-            if (!FileSystem.Current.DirectoryExists(fullsourceDirName))
+
+            // Windows will throw if the source file/directory doesn't exist, we preemptively check
+            // to make sure our cross platform behavior matches NetFX behavior.
+            if (!FileSystem.Current.DirectoryExists(fullsourceDirName) && !FileSystem.Current.FileExists(fullsourceDirName))
```
Comment (stephentoub) [inline src/System.IO.FileSystem/src/System/IO/Directory.cs]: It's unfortunate we're now making multiple calls to FillAttributeInfo on Windows and FileExists on Unix: each one appears to do the same thing but then checks whether the result is a directory or a file.  Should we add a DirectoryOrFileExists to the pal so that the implementation can do these checks once rather than twice?

Comment (JeremyKuhne) [inline src/System.IO.FileSystem/src/System/IO/Directory.cs]: Yes, we should. Created #19717 to track doing this post 2.0.

Comment (stephentoub) [inline src/System.IO.FileSystem/src/System/IO/Directory.cs]: Ok

File: src/System.IO.FileSystem/src/System/IO/DirectoryInfo.cs [hunk 2]
```diff
@@ -379,19 +379,20 @@ public DirectoryInfo Root
...
             if (destDirName.Length == 0)
                 throw new ArgumentException(SR.Argument_EmptyFileName, nameof(destDirName));
             Contract.EndContractBlock();
 
-            String fullDestDirName = Path.GetFullPath(destDirName);
-            if (fullDestDirName[fullDestDirName.Length - 1] != Path.DirectorySeparatorChar)
-                fullDestDirName = fullDestDirName + PathHelpers.DirectorySeparatorCharAsString;
+            string destination = Path.GetFullPath(destDirName);
+            string destinationWithSeparator = destination;
+            if (destinationWithSeparator[destinationWithSeparator.Length - 1] != Path.DirectorySeparatorChar)
```
Comment (stephentoub) [inline src/System.IO.FileSystem/src/System/IO/DirectoryInfo.cs]: On Window, what if it's the AltDirectorySeparatorChar?

Comment (JeremyKuhne) [inline src/System.IO.FileSystem/src/System/IO/DirectoryInfo.cs]: I'll add that case explicitly

File: src/System.IO.FileSystem/src/System/IO/Directory.cs (hunk 1 above)
Comment (danmosemsft) [inline src/System.IO.FileSystem/src/System/IO/Directory.cs]: We dont need API approval to add `FileSystem.PathExists` right now to make it efficient, but I'm fine of course with this as is.

Comment (JeremyKuhne) [inline src/System.IO.FileSystem/src/System/IO/Directory.cs]: @danmosemsft  I do want to make it better- I'm just a little worried about unnecessary changes this late. This stuff is pretty subtle. :)

Comment (danmosemsft) [inline src/System.IO.FileSystem/src/System/IO/Directory.cs]: Yep makes sense to me

File: src/System.IO.FileSystem/src/System/IO/UnixFileSystem.cs [hunk 3]
```diff
@@ -413,7 +407,9 @@ private static bool DirectoryExists(string fullPath, out Interop.ErrorInfo error
         public override bool FileExists(string fullPath)
         {
             Interop.ErrorInfo ignored;
-            return FileExists(fullPath, Interop.Sys.FileTypes.S_IFREG, out ignored);
+
+            // Windows doesn't care about the trailing separator, str
```
Comment (danmosemsft) [inline src/System.IO.FileSystem/src/System/IO/UnixFileSystem.cs]: "str"?

Comment (JeremyKuhne) [inline src/System.IO.FileSystem/src/System/IO/UnixFileSystem.cs]: Whoops- fixing
